src/plugin-runner.ts

```diff
@@ -0,0 +1,82 @@
+import type {FileSystem} from './file-system.js';
+import type {Message, Options, ReportPlugin, Stat, Stats} from './types.js';
+
+interface PluginTiming {
+  name: string;
+  ms: number;
+  error?: unknown;
+}
+
+interface RunPluginsResult {
+  messages: Message[];
+  stats: Stats;
+  timings: PluginTiming[];
+}
+
+function defaultStats(): Stats {
+  return {
+    name: 'unknown',
+    version: 'unknown',
+    dependencyCount: {
+      production: 0,
+      development: 0,
+      cjs: 0,
+      esm: 0,
+      duplicate: 0
+    },
+    extraStats: []
+  };
+}
+
+export async function runPlugins(
```

Comment:
i wonder if it should just mutate the stats we pass in instead of making its own copy

we're unnecessarily cloning stuff and what not. it should probably just mutate `messages` and `stats` we pass in